File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/base/BaseActivity.java
package com.example.stas.oxforddictionary.presentation.view.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.example.stas.oxforddictionary.presentation.navigation.Navigator;

public abstract class BaseActivity extends AppCompatActivity {

    protected Navigator navigator = new Navigator();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }
}

File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/base/BaseActivity.kt
package com.example.stas.oxforddictionary.presentation.view.base

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.inputmethod.InputMethodManager
import com.example.stas.oxforddictionary.presentation.navigation.Navigator

abstract class BaseActivity : AppCompatActivity() {

    protected var navigator = Navigator()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    fun closeKeyboard(){
        val view = currentFocus
        view.let {
            val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(view!!.windowToken, 0)
        }

    }
}
